app/services/fact_importer.rb

```diff
@@ -116,13 +116,18 @@ def update_facts
   end
 
   def normalize(facts)
+    fact_pairs = facts.map {|k, v| [k.to_s, v.to_s]}
+    # filter out matching fact names and empty values
+    fact_pairs.reject! {|k, v| v.empty? || k.match(ignored_fact_names)}
     # convert all structures to simple strings
-    facts = Hash[facts.map {|k, v| [k.to_s, v.to_s]}]
-    # and remove empty values
-    facts.keep_if { |k, v| v.present? }
+    Hash[fact_pairs]
   end
```

Comment:
I think it is a good opportunity to clean this method - this looks really ugly. Create pairs array, reject it, then create hash from it. I think creating copy of hash and then deleting filtered entries should be faster (leaving some buckets unused but faster).

    facts.select { |key, value| key.to_s.match(ignored_fact_names) }

This should do it, not tested tho.